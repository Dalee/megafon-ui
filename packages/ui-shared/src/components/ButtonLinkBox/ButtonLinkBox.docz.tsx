import * as React from 'react';

export const BackgroundWrapper: React.FC = ({ children }) => (
    <div style={{ background: '#00B956' }}>{children}</div>
);
