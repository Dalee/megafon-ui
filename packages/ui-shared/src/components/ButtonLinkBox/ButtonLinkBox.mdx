---
name: ButtonLinkBox
menu: Constructor
route: components/button-link-box
---

import { Playground, Props } from 'docz';
import ButtonLinkBox from './ButtonLinkBox';
import { BackgroundWrapper } from './ButtonLinkBox.docz';

# ButtonLinkBox

## Properties ButtonLinkBox
<Props of={ButtonLinkBox} />

## Basic usage

<Playground>
    <ButtonLinkBox
        buttonTitle="Button"
        buttonUrl="#"
        linkTitle="Link"
        linkUrl="#"
        className="custom-class"
    />
</Playground>

## Button only usage

<Playground>
    <ButtonLinkBox
        buttonTitle="Button"
        buttonUrl="#"
    />
</Playground>

## Link only usage

<Playground>
    <ButtonLinkBox
        linkTitle="Link"
        linkUrl="#"
    />
</Playground>

## Button color usage

<Playground>
    <ButtonLinkBox
        buttonTitle="Button"
        buttonUrl="#"
        buttonColor="purple"
    />
</Playground>

## Gorizontal alignment usage

<Playground>
    <ButtonLinkBox
        buttonTitle="Button"
        buttonUrl="#"
        linkTitle="Link"
        linkUrl="#"
        hAlign="center"
    />
</Playground>

## Terms of usage on a green background
##### - it's forrbidden to use link
##### - button is to be purple

<Playground>
    <BackgroundWrapper>
        <ButtonLinkBox
            buttonTitle="Button"
            buttonUrl="#"
            buttonColor="purple"
            hAlign="center"
        />
    </BackgroundWrapper>
</Playground>
