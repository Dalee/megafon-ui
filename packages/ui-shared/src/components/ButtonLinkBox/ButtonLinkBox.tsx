import * as React from 'react';
import * as PropTypes from 'prop-types';
import { cnCreate, Button, TextLink } from '@megafon/ui-core';
import './ButtonLinkBox.less';

export interface IButtonLinkBoxProps {
    className?: string;
    buttonTitle?: string;
    buttonUrl?: string;
    buttonColor?: 'green' | 'purple';
    linkTitle?: string;
    linkUrl?: string;
    hAlign?: 'center';
}

const cn = cnCreate('mfui-button-link-box');
const ButtonLinkBox: React.FC<IButtonLinkBoxProps> = ({
    buttonTitle,
    buttonUrl,
    buttonColor = 'green',
    linkTitle,
    linkUrl,
    hAlign,
    className,
}) => (
    <div className={cn({ 'h-align': hAlign }, className)}>
        {buttonTitle && (
            <div className={cn('row')}>
                <Button
                    href={buttonUrl}
                    passiveColor={buttonColor}
                    hoverColor={buttonColor}
                >
                    {buttonTitle}
                </Button>
            </div>
        )}
        {linkTitle && (
            <div className={cn('row')}>
                <TextLink href={linkUrl} underlineVisibility="always">
                    {linkTitle}
                </TextLink>
            </div>
        )}
    </div>
);

ButtonLinkBox.propTypes = {
    className: PropTypes.string,
    buttonTitle: PropTypes.string,
    buttonUrl: PropTypes.string,
    buttonColor: PropTypes.oneOf(['green', 'purple']),
    linkTitle: PropTypes.string,
    linkUrl: PropTypes.string,
    hAlign: PropTypes.oneOf(['center']),
};

ButtonLinkBox.defaultProps = {
    buttonColor: 'green',
};

export default ButtonLinkBox;
