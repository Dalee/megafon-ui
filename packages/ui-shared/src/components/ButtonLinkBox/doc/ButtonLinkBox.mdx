---
name: ButtonLinkBox
menu: Конструктор
route: components/button-link-box
---

import { Props } from 'docz';
import ButtonLinkBox from '../ButtonLinkBox';
import Example from './ButtonLinkBox.example.mdx'

# ButtonLinkBox

<DoczTabs>
    <Example />
    <Props of={ButtonLinkBox} />
</DoczTabs>
