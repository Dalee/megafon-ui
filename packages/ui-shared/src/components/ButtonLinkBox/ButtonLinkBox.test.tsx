import * as React from 'react';
import { shallow } from 'enzyme';
import ButtonLinkBox, { IButtonLinkBoxProps } from './ButtonLinkBox';

const props: IButtonLinkBoxProps = {
    className: 'custom-class',
    buttonTitle: 'button title',
    buttonUrl: 'button-url',
    buttonColor: 'purple',
    linkTitle: 'link title',
    linkUrl: 'link-url',
    hAlign: 'center',
};

describe('<ButtonLinkBox />', () => {
    it('renders ButtonLinkBox', () => {
        const wrapper = shallow(<ButtonLinkBox {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });
});
