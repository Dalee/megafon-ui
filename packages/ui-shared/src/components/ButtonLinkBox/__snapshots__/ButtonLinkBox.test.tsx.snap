// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonLinkBox /> renders ButtonLinkBox 1`] = `
<div
  className="mfui-button-link-box mfui-button-link-box_h-align_center custom-class"
>
  <div
    className="mfui-button-link-box__row"
  >
    <Button
      href="button-url"
      theme="purple"
    >
      button title
    </Button>
  </div>
  <div
    className="mfui-button-link-box__row"
  >
    <TextLink
      color="blue"
      href="link-url"
      underlineStyle="solid"
      underlineVisibility="always"
    >
      link title
    </TextLink>
  </div>
</div>
`;

exports[`<ButtonLinkBox /> renders ButtonLinkBox without button 1`] = `
<div
  className="mfui-button-link-box mfui-button-link-box_h-align_center custom-class"
>
  <div
    className="mfui-button-link-box__row"
  >
    <TextLink
      color="blue"
      href="link-url"
      underlineStyle="solid"
      underlineVisibility="always"
    >
      link title
    </TextLink>
  </div>
</div>
`;

exports[`<ButtonLinkBox /> renders ButtonLinkBox without link 1`] = `
<div
  className="mfui-button-link-box mfui-button-link-box_h-align_center custom-class"
>
  <div
    className="mfui-button-link-box__row"
  >
    <Button
      href="button-url"
      theme="purple"
    >
      button title
    </Button>
  </div>
</div>
`;
